src/string/immutable.zig

```diff
@@ -401,6 +401,12 @@ pub fn indexOfSigned(self: string, str: string) i32 {
     return @as(i32, @intCast(i));
 }
 
+/// Returns last index of `char` before a character `before`.
+pub fn lastIndexBeforeChar(self: string, char: u8, before: u8) ?usize {
```

Comment:
```suggestion
pub fn lastIndexBeforeChar(self: []const u8, char: u8, before: u8) ?usize {
```

We decided a little while ago to stop using the `string` alias in favor of `[]const u8`. Feel free to remove it from existing code you find. As you can see even this file doesn't use it consistently.

Reply:
Oh, we should stop naming the first parameter `self` too. I realize this isn't something you came up with as a bunch of functions in this file do it, but we might as well not keep doing it. It doesn't make any sense because you can't call any of these functions with method syntax on a `[]const u8`.